File: CampusView/backend/src/main/java/com/example/campusview/entity/ResourceBooking.java
package com.example.campusview.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "resource_bookings")
@Data
@NoArgsConstructor
public class ResourceBooking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String resourceType;

    @Column(nullable = false)
    private String resourceName;

    private String location;

    private Integer capacity;

    @Column(nullable = false)
    private LocalDateTime startTime;

    @Column(nullable = false)
    private LocalDateTime endTime;

    @Column(nullable = false)
    private Boolean isAvailable = true;

    private String description;

    public ResourceBooking(String resourceType, String resourceName) {
        this.resourceType = resourceType;
        this.resourceName = resourceName;
    }
} 

File: CampusView/app/src/main/java/com/example/campusview/data/entity/ResourceBooking.kt
package com.example.campusview.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

@Entity(tableName = "resource_booking")
data class ResourceBooking(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val resourceType: String, // '讲座' 或 '实验室'
    val resourceName: String,
    val location: String?,
    val capacity: Int?,
    val startTime: Date?,
    val endTime: Date?,
    val isAvailable: Boolean = true,
    val description: String?
) 